docs/navbar.js

```diff
@@ -1,77 +1,92 @@
+  try {
+    console.log('trying');
+    await privilegedUserPromise;
+  } catch (err) {
+    console.log('error');
+    if (err.code === AUTHENTICATION_ERROR_CODE) {
+      navPublic.show();
+    } else {
+      showError(err, 'Error populating navbar. Try refreshing page.');
+    }
+    return;
+  }
+  console.log('succeeded');
+  navToggle.show();
+  console.log($('#nav-toggle').is(':visible'));
+  $('#map-a').prop('href', getUrlUnderDocs(''));
+  $('#manage-layers-a').prop('href', getUrlUnderDocs(MANAGE_LAYERS_PAGE));
+  $('#manage-disaster-a').prop('href', getUrlUnderDocs(MANAGE_DISASTERS_PAGE));
+  $('#upload-image-collection-a')
+      .prop('href', getUrlUnderDocs(UPLOAD_IMAGE_COLLECTION_PAGE));
 }
 
 /**
  * Loads the navbar with a disaster picker.
- * @param {Promise} firebaseAuthPromise Promise that completes when Firebase
- *     login is done
- * @param {string} title Title of page
- * @param {?Function} changeDisasterHandler Function invoked when disaster is
- *     changed. If not specified, reloads page
- * @param {?Promise<Map<string, Object>>} firebaseDataPromise If caller has
- *     already kicked off a fetch of all disasters, pass that Promise in here to
- *     avoid a duplicate fetch
+ * @param {Object} data
  */
-function loadNavbarWithPicker(
-    firebaseAuthPromise, title, changeDisasterHandler = null,
-    firebaseDataPromise = null) {
+function loadNavbarWithPicker(data) {
+  let {
+    // promise that completes when Firebase login is done
+    firebaseAuthPromise,
+    // title of the page
+    title,
+    // function invoked when disaster is changed, if not specified, reloads page
+    changeDisasterHandler,
+    // fetch of all disasters
+    firebaseDataPromise,
+    // fetch of all user shapes - used as a proxy to determine if loading
+    // for a privileged user or not
+    userShapesPromise,
+  } = data;
   if (!firebaseDataPromise) {
     firebaseDataPromise = firebaseAuthPromise.then(getDisastersData);
   }
-  loadNavbar(() => {
-    $('#nav-left')
-        .load(
-            getUrlUnderDocs('disaster_picker.html'),
-            () => initializeDisasterPicker(
-                firebaseDataPromise, changeDisasterHandler));
-    $('#nav-title-header').html(title);
-  });
-}
-
-/**
- * Get url of a file in or below our docs directory by using the path of this
- * current script.
- * @param {string} pathFragment path fragment to append to '.../docs/'
- * @return {string}
- */
-function getUrlUnderDocs(pathFragment) {
-  return import.meta.url.replace(/navbar\.js$/, pathFragment);
-}
-
-/**
- * Gets the url for the help section relevant to the current page
- * @return {string}
- */
-function getHelpUrl() {
-  if (window.location.pathname.endsWith(MANAGE_LAYERS_PAGE)) {
-    return MANAGE_LAYERS_HELP_URL;
-  } else if (window.location.pathname.endsWith(MANAGE_DISASTERS_PAGE)) {
-    return MANAGE_DISASTERS_HELP_URL;
+  // On every page other than the main map, this will be null but if you're
```

Comment:
Maybe name this something more generic? Also, why not make this promise trivially resolve if caller passes in null, meaning they know it will succeed? (Or callers could pass in trivial promise themselves.)